lib/core/presentation/pages/chat/chat_list/views/widgets/guild_roles_list.dart

```diff
@@ -0,0 +1,162 @@
+import 'package:app/core/application/chat/check_guild_room_roles_bloc/check_guild_room_roles_bloc.dart';
+import 'package:app/core/domain/chat/entities/guild.dart';
+import 'package:app/core/domain/chat/entities/guild_room.dart';
+import 'package:app/core/presentation/pages/chat/chat_list/views/widgets/guild_detail_item.dart';
+import 'package:app/core/presentation/pages/chat/chat_list/views/widgets/guild_role_item.dart';
+import 'package:app/core/presentation/widgets/loading_widget.dart';
+import 'package:app/core/service/wallet/wallet_connect_service.dart';
+import 'package:app/core/utils/guild_utils.dart';
+import 'package:app/i18n/i18n.g.dart';
+import 'package:app/injection/register_module.dart';
+import 'package:app/router/app_router.gr.dart';
+import 'package:app/theme/color.dart';
+import 'package:app/theme/spacing.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class GuildRolesList extends StatelessWidget {
+  const GuildRolesList({super.key, required this.guildRoom});
+
+  final GuildRoom guildRoom;
+
+  @override
+  Widget build(BuildContext context) {
+    final t = Translations.of(context);
+    final colorScheme = Theme.of(context).colorScheme;
+
+    return Column(
+      children: [
+        BlocBuilder<CheckGuildRoomRolesBloc, CheckGuildRoomRolesState>(
+          builder: (context, state) {
+            return state.maybeWhen(
+              success: (guild, guildRoles, guildRolePermissions) {
+                // If didn't select any roles
+                if (guildRoom.guildRoleIds == null) {
+                  if (guild != null) {
+                    return _buildGuildDetailItem(
```

Comment:
Make this as a private widget inside this file only, not prefer to use render method here